persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/PersistentEntityRef.scala

```diff
@@ -37,14 +41,23 @@ final class PersistentEntityRef[Command](
   def ask[Cmd <: Command with PersistentEntity.ReplyType[_]](command: Cmd): Future[command.ReplyType] = {
     import scala.compat.java8.FutureConverters._
     import system.dispatcher
-    (region ? CommandEnvelope(entityId, command))
+    val result = (region ? CommandEnvelope(entityId, command))
       .flatMap {
         case exc: Throwable =>
           // not using akka.actor.Status.Failure because it is using Java serialization
           Future.failed(exc)
         case result => Future.successful(result)
       }
-      .asInstanceOf[Future[command.ReplyType]]
+
+    result.onComplete {
+      case Failure(ex) =>
+        logger.error(
+          s"${ex.getClass.getName} when sending command [${command.getClass}] to entity identified by [$entityId]"
+        )
+      case _ =>
+    }
+
+    result.asInstanceOf[Future[command.ReplyType]]
```

Comment:
Also, why use an `onComplete` and not _just_ add the log on the `case exc:Throwable` on the `flatMap` above?

(same applies on the java code)

Reply:
Because it only covers a `Success[Throwable]`, not a failed future.

Reply:
Not that this is flatMap is taking a successful future with a throwable and converting it to a failed future, `Future.failed(exc)`.

Reply:
🤦🏼‍♂️